spring-kafka/src/main/java/org/springframework/kafka/config/KafkaListenerEndpointRegistrar.java

```diff
@@ -74,6 +79,25 @@ public KafkaListenerEndpointRegistry getEndpointRegistry() {
 		return this.endpointRegistry;
 	}
 
+	/**
+	 * Return the list of {@link HandlerMethodArgumentResolver}.
+	 * @return the list of {@link HandlerMethodArgumentResolver}.
+	 * @since 2.5
+	 */
+	public List<HandlerMethodArgumentResolver> getCustomMethodArgumentResolvers() {
+		return this.customMethodArgumentResolvers;
```

Comment:
The good practice is to remove a `Collections.unmodifiableList()`

Reply:
I assume that you would like to write - The good practice is to **add** a Collections.unmodifiableList()

Reply:
Doh, of course! :shame:

Right, *wrap* an original `List` into a `Collections.unmodifiableList()` in getters to avoid unexpected external mutation.